modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/store/FilePageStore.java

```diff
@@ -0,0 +1,777 @@
+
+                        assert allocatedBytes.get() == 0;
+
+                        allocatedBytes.set(newSize);
+
+                        initialized = true;
+                    } catch (IOException e) {
+                        err = new IgniteInternalCheckedException("Failed to initialize partition file: " + filePath, e);
+
+                        throw err;
+                    } finally {
+                        if (err != null && fileIo != null) {
+                            try {
+                                fileIo.close();
+                            } catch (IOException e) {
+                                err.addSuppressed(e);
+                            }
+                        }
+                    }
+                }
+            } finally {
+                readWriteLock.writeLock().unlock();
+            }
+        }
+    }
+
+    /**
+     * Initializes header and writes it into the file store.
+     *
+     * @return Next available position in the file to store a data.
+     * @throws IOException If initialization is failed.
+     */
+    private long initFile(FileIo fileIo) throws IOException {
+        try {
+            ByteBuffer hdr = header(type, pageSize);
+
+            fileIo.writeFully(hdr);
+
+            // There is 'super' page in every file.
+            return 0;
+        } catch (ClosedByInterruptException e) {
+            // If thread was interrupted written header can be inconsistent.
+            readWriteLock.writeLock().lock();
+
+            try {
+                Files.delete(filePath);
+
+                fileExists = false;
+            } finally {
+                readWriteLock.writeLock().unlock();
+            }
+
+            throw e;
+        }
+    }
+
+    /**
+     * Creates header for current version file store.
+     *
+     * <p>Doesn't init the store.
+     *
+     * @param type Data type, can be {@link PageStore#TYPE_IDX} or {@link PageStore#TYPE_DATA}.
+     * @param pageSize Page size in bytes.
+     * @return Byte buffer instance.
+     */
+    private ByteBuffer header(byte type, int pageSize) {
+        ByteBuffer hdr = ByteBuffer.allocate(headerSize()).order(nativeOrder());
+
+        hdr.putLong(SIGNATURE);
+
+        hdr.putInt(version());
+
+        hdr.put(type);
+
+        hdr.putInt(pageSize);
+
+        hdr.rewind();
+
+        return hdr;
+    }
+
+    /**
+     * Checks that file store has correct header and size.
+     *
+     * @return Next available position in the file to store a data.
+     * @throws IOException If check has failed.
+     */
+    private long checkFile(FileIo fileIo, Path filePath) throws IOException {
+        ByteBuffer headerBuffer = ByteBuffer.allocate(headerSize()).order(nativeOrder());
+
+        fileIo.readFully(headerBuffer);
+
+        checkHeader(headerBuffer);
+
+        long fileSize = filePath.toFile().length();
+
+        // Every file has a special meta page.
+        if (fileSize == headerSize()) {
```

Comment:
This whole "if" can just be replaces with "fileSize -= headerSize();"

Reply:
Fix it